lib/swaps/SwapClient.ts

```diff
@@ -142,43 +143,72 @@ abstract class SwapClient extends EventEmitter {
     });
   }
 
-  protected setStatus = async (status: ClientStatus): Promise<void> => {
+  public init = async () => {
+    // up front checks before initializing client
+    if (!this.isNotInitialized() && !this.isMisconfigured()) {
+      // we only initialize from NotInitialized or Misconfigured status
+      this.logger.warn(`can not init in ${this.status} status`);
+      return;
+    }
+    if (this.disable) {
+      this.setStatus(ClientStatus.Disabled);
+      return;
+    }
+
+    // client specific initialization
+    await this.initSpecific();
+
+    // final steps to complete initialization
+    this.setStatus(ClientStatus.Initialized);
+    this.setTimers();
+    this.emit('initialized');
+    await this.verifyConnectionWithTimeout();
+  }
+
+  protected abstract async initSpecific(): Promise<void>;
+
+  protected setConnected = async (newPubKey?: string, newUris?: string[]) => {
+    await this.updateCapacity();
+    this.setStatus(ClientStatus.ConnectionVerified);
+    this.emit('connectionVerified', {
+      newUris,
+      newIdentifier: newPubKey,
+    });
+
+  }
+
+  protected setStatus = (status: ClientStatus): void => {
     if (this.status === status) {
       return;
     }
 
     this.logger.info(`${this.constructor.name} status: ${ClientStatus[status]}`);
     this.status = status;
-    await this.setTimers();
   }
 
-  private setTimers = async () => {
-    if (this.status === ClientStatus.ConnectionVerified) {
-      if (!this.updateCapacityTimer) {
-        await this.updateCapacity();
-        this.updateCapacityTimer = setInterval(this.updateCapacity, SwapClient.CAPACITY_REFRESH_INTERVAL);
-      }
-
-      if (this.reconnectionTimer) {
-        clearTimeout(this.reconnectionTimer);
-        this.reconnectionTimer = undefined;
-      }
-    } else {
-      if (this.updateCapacityTimer) {
-        clearInterval(this.updateCapacityTimer);
-        this.updateCapacityTimer = undefined;
-      }
-      if (this.status === ClientStatus.Disconnected || this.status === ClientStatus.OutOfSync || this.status === ClientStatus.WaitingUnlock) {
-        if (!this.reconnectionTimer) {
-          this.reconnectionTimer = setTimeout(async () => {
-            await this.verifyConnection();
-            if (!this.isConnected() && this.reconnectionTimer) {
-              // if we were still not able to verify the connection, schedule another attempt
-              this.reconnectionTimer.refresh();
-            }
-          }, SwapClient.RECONNECT_TIME_LIMIT);
-        }
-      }
+  private updateCapacityTimerCallback = async () => {
+    if (this.isConnected()) {
+      await this.updateCapacity();
+    }
+  }
+
+  private reconnectionTimerCallback = async () => {
+    if (this.status === ClientStatus.Disconnected || this.status === ClientStatus.OutOfSync || this.status === ClientStatus.WaitingUnlock) {
+      try {
+        await this.verifyConnection();
+      } catch {}
```

Comment:
Are we sure about silencing all errors here? Maybe `.catch(this.logger.error)` or `this.logger.trace`?

Reply:
I did think about this some and wasn't too sure, I can put in a debug line. I think any errors are already already being logged elsewhere so I didn't want to pollute the logs.